### PR TITLE
[Code] [Networking] GIga Rework: Switch to Photon PUN

### PR DESCRIPTION
**Pull request by KristianKurk**
fuck you mirror, we're photon gang now

peep the latest commit, it's the relevant one. Everything else is setting up the environment and deleting bloat files.

### CODE REVIEW

File: Project Bunny/Assets/Scripts/Player/NetworkStudentController.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        [SerializeField] private CharacterController _characterController;
+        [SerializeField] private float _movementSpeed;
+
+        private PhotonView view;
+        private Vector3 _currentPosition;
+
+        private void Awake()
+        {
+            if (_characterController == null)
+                _characterController = GetComponent<CharacterController>();
```
Comment (George540) [inline Project Bunny/Assets/Scripts/Player/NetworkStudentController.cs]: Please use brackets to close if statements like this
```
if (_characterController == null)
{
                _characterController = GetComponent<CharacterController>();
}
```

File: Project Bunny/Assets/Scripts/Player/NetworkStudentController.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            if (view == null)
+                view = GetComponent<PhotonView>();
+
+            GetComponent<PlayerInput>().actionEvents[0].AddListener(OnMove);
+            GetComponent<PlayerInput>().actionEvents[1].AddListener(OnLook);
+        }
+
+        private void FixedUpdate()
+        {
+            if (!view.IsMine) return;
```
Comment (George540) [inline Project Bunny/Assets/Scripts/Player/NetworkStudentController.cs]: for returns and continues, it's not necessary to do brackets

**Review by George540 — APPROVED** [review]
Good good

**Review by DrakeHunterK7 — APPROVED** [review]
Code looks great and seems like everything is in. 
Genius code indeed.

**Pull request merged**